GEODE-4881: Support lucene reindexing (of existing data) with rebalance

ladyVader opened the pull request:
* LuceneServiceImpl changes to support AEQ addition to existing region (after index initialized)
* Testing with and without redundancy

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

nabarunnag commented on geode-lucene/src/test/java/org/apache/geode/cache/lucene/RebalanceWithRedundancyWithRegionCreatedBeforeReindexDUnitTest.java:
```diff
@@ -0,0 +1,343 @@
+      try {
+        rebalanceResults = rebalanceOp.getResults();
+      } catch (InterruptedException e) {
+        e.printStackTrace();
```
Should we fail the test if re-balance encountered an Exception ??

nabarunnag replied:
maybe re-throw exception so that it caught in the main tests checkException() method.

ladyVader replied:
Good catch, thanks!

ladyVader commented:
Updated by allowing InterruptedException to be thrown back to caller.

In addition, updated the Test Category to include LuceneTest and used logger to display rebalance activities.
